### ISSUE TITLE
Inconsistent expiration behaviour for memory storage when replacing

### ISSUE THREAD

**Issue by instanceofme** [mikehostetler/amplify#23]
With`*Storage` and `userData`, if I replace a previously-expirable value and don't set an expiration, there is no expiration (which is the behavior I expect). With `memory` storage however, the first expiration will delete the replaced value (the `setTimeout` is not cleared).

```
amplify.store[type]( 'key', 'value1', {expires: 500} );
amplify.store[type]( 'key', 'value2' );
window.setTimeout( function(){ alert( amplify.store[type]( 'key' ) ) }, 1000 );
```

[This JsFiddle](http://jsfiddle.net/QvUth/) tests all storage types available to the current browser.

**Comment by instanceofme** [comment]
Fix proposal in [pull request #24](https://github.com/appendto/amplify/pull/24), I don't know if you can merge those discussions…

**Comment by scottgonzalez** [comment]
I didn't want that commit to close this ticket. I just wanted to post my suggested fix to start a discussion with @streetpc. I guess next time I'll have to do it from my own account...
